src/platform/Darwin/DnssdContexts.cpp

```diff
@@ -548,7 +548,7 @@ void ResolveContext::DispatchSuccess()
 
     for (auto & interface : interfaces)
     {
-        if (TryReportingResultsForInterfaceIndex(interface.first))
+        if (TryReportingResultsForInterfaceIndex(interface.first.first, interface.first.second))
```

Comment:
I think this code would be more readable if we had structs with 2 meaningfully-named fields, rather than using std::pair. If you use aggregate initialization it doesn't involve writing more code either.

Reply:
I used std::pair from my conversation with Boris. I could add comments in code to say what first and second are if that helps.

Reply:
I have to agree with Karsten here. Use a #define if you need to. :)

Reply:
for now I am going to add variables where i get the values from the pair with proper naming so there's no confusion. I need to get the addressed comments merged today. Please let me know if you still feel strongly about it.

Reply:
so I tried to change this to a struct and running into issues where the map does some sorting based on keys and is looking for some operators on the struct. I spent some time adding < and + operator but seems like a non trivial fix here.

Reply:
Filed : #32678